src/html_tokenizer.ml

```diff
@@ -375,7 +375,7 @@ let tokenize report (input, get_location) =
               | Trie.No ->
                 finish best matched (v::replace)
               | Trie.Prefix ->
-                match_named best matched (v::replace) trie text
+                match_named best (v::matched) (v::replace) trie text
```

Comment:
I still haven't had time to take a proper look (sorry!). However, what immediately jumps out at me about this way of fixing it, is that on line 381, `replace` and `matched` are concatenated. This makes me suspect that it's wrong to add `v` to both.

A bit embarrassed here, as I need to relearn my own code :/ (and I'll be commenting and/or clarifying these functions after this PR).